### PR TITLE
[libjuice] update to 1.0.0

### LINKED ISSUE #24284: [libjuice] update to 1.0.0
Library name: libjuice

New version number: 1.0.0

https://github.com/paullouisageneau/libjuice/releases

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #24284

Update libjuice to the latest version 1.0.0

Feature tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

### CODE REVIEW

File: ports/libjuice/vcpkg.json [hunk 1]
```diff
@@ -1,8 +1,9 @@
 {
   "name": "libjuice",
-  "version": "0.9.8",
+  "version": "1.0.0",
   "description": "The library is a simplified implementation of the Interactive Connectivity Establishment (ICE) protocol in C for POSIX platforms (including Linux and Apple macOS) and Microsoft Windows.",
   "homepage": "https://github.com/paullouisageneau/libjuice",
+  "license": "LGPL-2.1",
```
Comment (FrankXie05) [inline ports/libjuice/vcpkg.json]:
```suggestion
  "license": "LGPL-2.1",
```
`LGPL-2.1`- only or later?

**Review by FrankXie05 — APPROVED** [review]

**Comment by PriitUring** [comment]
@FrankXie05
> Update libjuice to the latest version 1.0.1

Typo, opening post should have written "version 1.0.0" not something that's unreleased "version 1.0.1".

**Comment by FrankXie05** [comment]
@PriitUring Ah, it's my fault, thanks for reminding. :)

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**